### PR TITLE
Add ghost circuit for Input buses

### PR DESCRIPTION
**Pull request by lukaslcf**
as title

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java [hunk 1]
```diff
@@ -169,6 +191,13 @@ public Widget createUIWidget() {
 
         container.setBackground(GuiTextures.BACKGROUND_INVERSE);
         group.addWidget(container);
+
+        if (this.io == IO.IN) {
+            var configurationPanel = new ConfiguratorPanel();
+            this.attachConfigurators(configurationPanel);
+
+            group.addWidget(configurationPanel);
+        }
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java:200]: remove it

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java [hunk 2]
```diff
@@ -149,6 +166,11 @@ public void setWorkingEnabled(boolean workingEnabled) {
     //////////////////////////////////////
     //**********     GUI     ***********//
     //////////////////////////////////////
+
+    public void attachConfigurators(ConfiguratorPanel configuratorPanel) {
+        configuratorPanel.attachConfigurators(new CircuitFancyConfigurator(circuitInventory.storage));
+    }
+
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java:173]: super.attachConfugurator(configuratorPanel)
if (this.io == IO.IN) {
     configuratorPanel.attachConfigurators(new CircuitFancyConfigurator(circuitInventory.storage));
}

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java [hunk 3]
```diff
@@ -68,9 +77,17 @@ protected NotifiableItemStackHandler createInventory(Object... args) {
         return new NotifiableItemStackHandler(this, getInventorySize(), io);
     }
 
+    protected NotifiableItemStackHandler createCircuitItemHandler(Object... args) {
+        return new NotifiableItemStackHandler(this, 1, IO.IN).setFilter(IntCircuitBehaviour::isIntegratedCircuit);
+    }
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java:82]: if (arg[]s.length > 0 && args[0] instanceof IO io && io == IO.IN){
    return new NotifiableItemStackHandler(this, 1, IO.IN).setFilter(IntCircuitBehaviour::isIntegratedCircuit);
} else {
    return new NotifiableItemStackHandler(this, 0, IO.NONE);
}

File: common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java [hunk 4]
```diff
@@ -45,10 +50,14 @@ public class ItemBusPartMachine extends TieredIOPartMachine implements IDistinct
...
     @Nullable
     protected ISubscription inventorySubs;
+    @Getter
+    @Persisted
+    protected final NotifiableItemStackHandler circuitInventory;
 
     public ItemBusPartMachine(IMachineBlockEntity holder, int tier, IO io, Object... args) {
         super(holder, tier, io);
         this.inventory = createInventory(args);
+        this.circuitInventory = createCircuitItemHandler(args);
```
Comment (Yefancy) [inline common/src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/ItemBusPartMachine.java:60]: this.circuitInventory = createCircuitItemHandler(io);

**Review by Yefancy — CHANGES_REQUESTED** [review]

**Comment by lukaslcf** [comment]
All reviews noted, I will implement them and create a new PR from a new branch. Should I merge it to 1.19.2?

**Comment by screret** [comment]
why a new PR?

**Comment by lukaslcf** [comment]
> why a new PR?

I was thinking of PR-ing it into 1.19.2 instead of the 2.7 branch

**Comment by screret** [comment]
ah. makes sense.

**Pull request closed without merging**